monai/data/samplers.py

```diff
@@ -0,0 +1,93 @@
+# Copyright 2020 - 2021 MONAI Consortium
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Optional, Sequence
+
+import torch
+from torch.utils.data import DistributedSampler as _TorchDistributedSampler
+
+__all__ = ["DistributedSampler", "DistributedWeightedRandomSampler"]
+
+
+class DistributedSampler(_TorchDistributedSampler):
+    """
+    Enhance PyTorch DistributedSampler to support non-evenly divisible sampling.
+
+    Args:
+        even_divisible: if False, different ranks can have different data length.
+        for example, input data: [1, 2, 3, 4, 5], rank 0: [1, 3, 5], rank 1: [2, 4].
+        args: additional arguments for `DistributedSampler` super class.
+        kwargs: additional arguments for `DistributedSampler` super class.
+
+    More information about DistributedSampler, please check:
+    https://github.com/pytorch/pytorch/blob/master/torch/utils/data/distributed.py
+
+    """
+
+    def __init__(self, even_divisible: bool = True, *args, **kwargs):
```

Comment:
The common arguments between versions 1.5 and 1.8 are `dataset`, `num_replicas`, `rank`, `shuffle`. I think these should be declared in this constructor and `**kwargs` used to pass `seed` and `drop_last` if these are to be used with a note in the docstring stating that's what it's there for. We can drop the use of `*args` this way.

Reply:
Sure, I updated the PR according to your comments.
Thanks.